### PR TITLE
Parsing exceptions during the parsing of invalid SARIF result

### LINKED ISSUE #388: Parsing exceptions during the parsing of invalid SARIF result
If the tool provides us invalid SARIF result and we fail on parsing of it - we know it definitely.

We need to handle that situation and NOT crush save-cli with exception. Instead - we need to catch an error and mark all tests in the batch **FAILED** with the reason in the debug info: `WE FAILED TO PARSE SARIF!!!111111 WOW, YOUR TOOL IS INVALID, CHECK THE GENERATION OF SARIF REPORT`.

### PR DESCRIPTION
**Pull request by Cheshiriks**
* Added exception for parsing sarif
Closes #388

### CODE REVIEW

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt [hunk 1]
```diff
@@ -206,14 +219,19 @@ class WarnPlugin(
...
         warnPluginConfig: WarnPluginConfig,
         originalPaths: List<Path>,
         copyPaths: List<Path>,
         workingDirectory: Path,
     ): WarningMap = if (warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF) {
-        val warningsFromSarif = collectWarningsFromSarif(warnPluginConfig, originalPaths, fs, workingDirectory)
+        val warningsFromSarif = try {
+            collectWarningsFromSarif(warnPluginConfig, originalPaths, fs, workingDirectory)
+        } catch (e: Exception) {
+            throw ProcessParseSarifException("We failed to parse sarif. Your tool is invalid, check the generation of sarif report")
```
Comment (petertrr) [inline save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt:233]: Why are you ignoring the original exception? It contains very valuable information for the user. Was it deserialization exception, i.e. SARIF is malformed? Or was it an IO exception, e.g. file couldn't be opened for some reason? Please include it as a cause of this exception and check whether we discard it sometime later. Users should be able to see this information.
Also - wording `you tool is invalid` is too harsh :)

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/ProcessParseSarifException.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.cqfn.save.core.utils
+
+/**
+ * An [Exception] that can be thrown in case of failed to parse sarif
+ */
+open class ProcessParseSarifException(message: String) : Exception(message)
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/ProcessParseSarifException.kt:6]:
```suggestion
open class SarifParsingException(message: String) : Exception(message)
```
Also, why `open class`?

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt]: Ok. Fixed:)

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/ProcessParseSarifException.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/ProcessParseSarifException.kt]: Fixed

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/SarifParsingException.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.cqfn.save.core.utils
+
+/**
+ * An [Exception] that can be thrown in case of failed to parse sarif
+ */
+class SarifParsingException(message: String) : Exception(message)
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/SarifParsingException.kt:6]: Please also add `cause` as a parameter (`Exception` has a constructor that accepts cause). We may want to use it in the future for better reporting, so it'll be useful to have it in there already

Comment (Cheshiriks) [reply save-common/src/commonMain/kotlin/org/cqfn/save/core/utils/SarifParsingException.kt]: Added

**Pull request merged**